lrtable/src/lib/statetable.rs

```diff
@@ -551,6 +577,8 @@ fn resolve_shift_reduce<StorageT: 'static + Hash + PrimInt + Unsigned>(
             // If the token and production don't both have precedences, we use Yacc's default
             // resolution, which is in favour of the shift.
             actions[off] = StateTable::encode(Action::Shift(stidx));
+            let ridx = grm.prod_to_rule(pidx);
+            *rule_usage.get_mut(usize::from(ridx)).unwrap() = (ridx, true);
```

Comment:
I believe this marking things as used when there is a conflict resolved in their favor may not be right,
in cases when there is some subsequent conflict with the production that it was resolved in favor of

Reply:
I must admit that I have no idea!

Reply:
Well, I'm going to try and avoid relying on that, if we look at what is still reachable after conflicts have been removed.
and use the forward pass merely for rules which are plain unused. It seems like a less fragile invariant, and the actual problem I was running into with the case above, is the wrong `pidx`, and it seems much more difficult (if actually possible) to go from `stidx` back to a `pidx`

Reply:
From memory, a `StIdx` can represent multiple productions. I'm not sure if that helps you either way!